app/views/advanced_messenger/_html_head.html.erb

```diff
@@ -1,11 +1,20 @@
 <%# ============= Plugin styles  ============= %>
 <% content_for :header_tags do %>
     <%= stylesheet_link_tag 'advanced_messenger', :plugin => 'redmine_advanced_messenger' %>
+    <%# ============= PWA Manifest   ============= %>
+    <%= auto_discovery_link_tag :json, '../../plugin_assets/redmine_advanced_messenger/pwa/manifest.json', { rel: "manifest" } %>
 <% end %>  
+<%# ============= Plugin JS script  ============= %>
+<%= javascript_include_tag 'redmine_advanced_messenger', plugin: 'redmine_advanced_messenger' %>
+<%# ============= Register PWA service worker  ============= %>
+<%= javascript_tag do%>
+    registerPWAServiceWorker();
```

Comment:
I'm wondering if you can put here a check for whether or not you are running in a PWA and call the `registerPWAServiceWorker` only on PWA.

Reply:
Same thing as above.